### ISSUE TITLE
An exceptionCaught() event was fired, and it reached at the tail of the pipeline

### ISSUE THREAD

**Issue by CB-andrewehrlich** [zio/zio-http#1982]
This is running on `Ubuntu 20.04.4 LTS` on google cloud Intel Cascade Lake `n2-highcpu-32`

The server seems to issue warnings, repeatedly, all the time. The logs contain tons of these message:

```
io.netty.channel.unix.Errors$NativeIoException: recvAddress(..) failed: Connection reset by peer
2023-02-07 13:55:08,303 [epollEventLoopGroup-2-4] WARN  io.netty.channel.DefaultChannelPipeline - An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
```

This is a pretty small http server with just one route. Any suggestions for how to narrow the problem? I can post results here.

**Steps to reproduce:**

`Main.scala`
```
package com.example

import zio._
import zio.http._
import zio.http.model._

object Main extends ZIOAppDefault {

  val app: HttpApp[Any, Nothing] = Http.collectZIO {
    case req @ Method.GET -> !! / "health" => health(req)
  }

  def health(req: Request) = {
    println(req)
    val response = for {
      res <- ZIO.succeed("OK")
    } yield Response.json(res)
    response
  }

  private val httpConfig = ServerConfig.default

  override val run: ZIO[Any, Nothing, Unit] =
    Server.serve(app)
      .provide(
        ServerConfig.live(httpConfig),
        Server.live,
      ).exit.flatMap({
        case Exit.Success(_)     =>
          println("Exiting successfully...")
          ZIO.unit
        case Exit.Failure(cause) =>
          println("There were error(s) when running the program.")
          cause.failures.foreach(println)
          ZIO.unit
      })

}
```

`build.sbt`
```
val zioVersion            = "2.0.8"
val zioHttpVersion        = "0.0.4"

lazy val root = (project in file("."))
  .settings(
    inThisBuild(
      List(
        organization := "com.example",
        scalaVersion := "2.13.8",
      )
    ),
    name           := "zio-scala3-quickstart",
    libraryDependencies ++= Seq(
      "dev.zio"       %% "zio"                 % zioVersion,
      "dev.zio"       %% "zio-http"            % zioHttpVersion,
    ),
    testFrameworks := Seq(new TestFramework("zio.test.sbt.ZTestFramework")),
  )
  .enablePlugins(JavaAppPackaging)
  ```

**Comment by CB-andrewehrlich** [comment]
Actually, I confirmed that this happens on 0.0.3 too.

**Comment by Wosin** [comment]
Can You share the code that causes the issue ?

**Comment by CB-andrewehrlich** [comment]
@Wosin i updated the body of this ticket with code to reproduce

**Comment by jdegoes** [comment]
Closed in favor of #2071
